src/App.tsx

```diff
@@ -36,60 +36,107 @@ export function getReorderedGoods(
 
   // Sort and reverse goods if needed
   // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  if (sortType !== SortType.NONE) {
+    visibleGoods.sort((firstGood, secondGood) => {
+      if (sortType === SortType.ALPHABET) {
+        return firstGood.localeCompare(secondGood);
+      }
+
+      if (sortType === SortType.LENGTH) {
+        return firstGood.length - secondGood.length;
+      }
+
+      return 0; // this is to make linter shut the hell up
+    });
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleButton(actionType: string) {
+    if (actionType === 'reverse') {
+      this.setState((state) => ({ isReversed: !state.isReversed }));
+    }
+
+    if (actionType === 'reset') {
+      this.setState({ isReversed: false, sortType: SortType.NONE });
+    }
+
+    if (actionType === 'sortAlphabet') {
+      this.setState({ sortType: SortType.ALPHABET });
+    }
+
+    if (actionType === 'sortLength') {
+      this.setState({ sortType: SortType.LENGTH });
+    }
+  }
```

Comment:
maybe instead of so many if statements lets separate each of them to something like:
```
  reverseOrder = () => {
    this.setState(state => ({
      isReversed: !state.isReversed,
    }));
  };
```
and then desctructure it and use it in button's onClick: onClick={reverseOrder}

Reply:
With onClick={reverseOrder} and the like, the list does not update
Using onClick={() => (reverseOrder())} instead works